ProcessLib/ThermoHydroMechanics/CreateThermoHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,310 @@
+    if (use_monolithic_scheme)  // monolithic scheme.
+    {
+        auto per_process_variables = findProcessVariables(
+            variables, pv_config,
+            {//! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables__temperature}
+             "temperature",
+             //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables__pressure}
+             "pressure",
+             //! \ogs_file_param_special{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables__displacement}
+             "displacement"});
+        variable_T = &per_process_variables[0].get();
+        variable_p = &per_process_variables[1].get();
+        variable_u = &per_process_variables[2].get();
+        process_variables.push_back(std::move(per_process_variables));
+    }
+    else  // staggered scheme.
+    {
+        using namespace std::string_literals;
+        for (auto const& variable_name :
+             {"Temperature"s, "pressure"s, "displacement"s})
+        {
+            auto per_process_variables =
+                findProcessVariables(variables, pv_config, {variable_name});
+            process_variables.push_back(std::move(per_process_variables));
+        }
+        variable_T = &process_variables[0][0].get();
+        variable_p = &process_variables[1][0].get();
+        variable_u = &process_variables[2][0].get();
+    }
+
+    DBUG("Associate displacement with process variable \'%s\'.",
+         variable_u->getName().c_str());
+
+    if (variable_u->getNumberOfComponents() != DisplacementDim)
+    {
+        OGS_FATAL(
+            "Number of components of the process variable '%s' is different "
+            "from the displacement dimension: got %d, expected %d",
+            variable_u->getName().c_str(),
+            variable_u->getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    DBUG("Associate pressure with process variable \'%s\'.",
+         variable_p->getName().c_str());
+    if (variable_p->getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "Pressure process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            variable_p->getName().c_str(),
+            variable_p->getNumberOfComponents());
+    }
+
+    DBUG("Associate temperature with process variable \'%s\'.",
+         variable_T->getName().c_str());
+    if (variable_T->getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "temperature process variable '%s' is not a scalar variable but "
+            "has "
+            "%d components.",
+            variable_T->getName().c_str(),
+            variable_T->getNumberOfComponents());
+    }
+
+    auto solid_constitutive_relations =
+        MaterialLib::Solids::createConstitutiveRelations<DisplacementDim>(
+            parameters, config);
+
+    // Intrinsic permeability
+    auto& intrinsic_permeability = findParameter<double>(
```

Comment:
Yes, here is the simplified case with isotropic intrinsic permeability. I have added the comment on it.

Reply:
Since you use parameter for the permeability values, it is easy to change it to tensor with Dima's new implementation
(https://github.com/ufz/ogs/pull/2370/commits/0212c7f668fd4cd543b748586874b179c1749cbe). The same for fluid_thermal_conductivity. To avoid code duplication, here is a suggestion: 
1. leave a comment above  `double const K_over_mu =` as
    // TODO: change it like ``auto const  K_over_mu  = hydraulicConductivity<GlobalDim>(_process_data.intrinsic_permeability(t, x_position)/...``
2. leave a similar comment above    `auto const lambda`

Reply:
No problem. I have added the comments there !